src/Microsoft.Owin.Security.Authorization/IAuthorizationDependenciesProvider.cs

```diff
@@ -0,0 +1,11 @@
+using System;
+
+namespace Microsoft.Owin.Security.Authorization
+{
+    public interface IAuthorizationDependenciesProvider
+    {
+        Func<AuthorizationOptions, IOwinContext, AuthorizationDependencies> OnCreate { get; }
+
+        Action<AuthorizationOptions, IOwinContext, AuthorizationDependencies> OnDispose { get; }
```

Comment:
in my opinion request lifetime scoped handlers (which are injected in the authzservice) will often be disposable (loading e.g. permissions from a database) and it can't be the  middleware's responsibility to handle the disposals: using a container the container takes care of it, without a container (by creating disposable objects during the OnCreate call) it is the app's responsibility - the middleware can't differentiate! or do I miss something?